source/MaterialXGraphEditor/Main.cpp

```diff
@@ -253,7 +255,11 @@ int main(int argc, char* const argv[])
         ImGui_ImplGlfw_NewFrame();
         ImGui::NewFrame();
 
-        graph->getRenderer()->drawContents();
+        auto renderer = graph->getRenderer();
+        using FrameType = std::invoke_result_t<decltype(&RenderView::getFrame), RenderView>;
+        constexpr auto frameMaxValue = std::numeric_limits<FrameType>::max();
```

Comment:
This is a really interesting approach, and I especially like the one-line version of this logic that you use in Viewer.cpp below:

```
constexpr auto frameMaxValue = std::numeric_limits<decltype(_renderPipeline->_frame)>::max();
```

Is there a way to maintain the same, simpler approach in Main.cpp as well, rather than bringing in the more complex machinery of `std::invoke_result_t` from `type_traits`?

Reply:
In Viewer.cpp the _frame variable is part of the class using it, whereas here we don't have direct access to it as it is a private variable of another class. I didn't want to make the variable public just for this. Unless there is a solution I am missing?
I don't think the longer code here is a performance issue at least, as it should all be resolved at compile time. If it's too confusing I suppose we could simply use the type explicitly?

Reply:
Yes, it's more the additional complexity for future readers that I'm concerned with, as `std::invoke_result_t` is such an unusual technique, and `type_traits` is not otherwise used in this source file.

I like your suggestion to use the type explicitly, which would give the following one-line implementation:

```
constexpr auto FRAME_MAX_VALUE = std::numeric_limits<unsigned int>::max();
```

Note that I've also updated the name of the constant from `frameMaxValue` to `FRAME_MAX_VALUE`, in order to better match the coding conventions in MaterialX.

Does that approach seem reasonable to you as well?

Reply:
Sure, made the change.